pyproject.toml

```diff
@@ -1,4 +1,107 @@
+[tool.poetry]
+name = "dowhy"
+version = "0.8.0"
+description = "DoWhy is a Python library for causal inference that supports explicit modeling and testing of causal assumptions"
+authors = [
+    "Amit Sharma <amshar@microsoft.com>",
+    "Emre Kiciman <emrek@microsoft.com>"
+]
+maintainers = []
+license = "MIT"
+documentation = "https://py-why.github.io/dowhy"
+repository = "https://github.com/pywhy/dowhy"
+classifiers = [  
+    'Development Status :: 4 - Beta',
+    'License :: OSI Approved :: MIT License',
+    'Programming Language :: Python :: 3.7',
+    'Programming Language :: Python :: 3.8',
+]
+keywords = [
+    'causality',
+    'machine-learning',
+    'causal-inference',
+    'statistics',
+    'graphical-model'
+]
+exclude = ['docs', 'test']
+readme = 'README.rst'
+
+[tool.poetry.dependencies]
+python = ">=3.8,<3.11"
+scipy = "^1.8.1"
+statsmodels = "^0.13.2"
+numpy = "^1.23.1"
+pandas = "^1.4.3"
+networkx = "^2.8.5"
+sympy = "^1.10.1"
+scikit-learn = "1.0.2"
+pydot = "^1.4.2"
+joblib = "^1.1.0"
+pandas-stubs = {version = "^1.4.3", optional = true}
+llvmlite = "^0.31.0"
+matplotlib = {version = "^3.5.2", optional = true}
+# TODO: resolve circular dependency w/ econml
```

Comment:
the way we have done it so far is by not including econml in requirements.txt, but only in requirements-test.txt

Reply:
Poetry is setting up its own venv, and using a pip requirements file didn't work. Including it here this way fixed some tests, but I believe it's resolving to an older version without the circular dep.

If any dowhy library consumers (who also use poetry or conda) try to use econml, the circular dependency present will force them into a version < 0.8. This is something we should address with the econml folks over the long run - I think there's a fundamental versioning ergonomics issue here.